contracts/SportMarkets/LiquidityPool/SportAMMLiquidityPool.sol

```diff
@@ -88,6 +88,9 @@ contract SportAMMLiquidityPool is Initializable, ProxyOwned, PausableUpgradeable
 
     /* ========== CONSTRUCTOR ========== */
 
+    // Initializes the implementation contract to avoid potential foul-play
+    constructor() initializer {}
```

Comment:
are we sure that this wont block the actual initialize call on first deployment of the proxy?

Reply:
@danijelthales I'm sure ser. No need to take my word for it though, I wrote a couple tests so you can verify: https://github.com/kphed/proxy-demo/blob/master/test/Counter.t.sol#L37.

Reply:
> If _data is nonempty, it’s used as data in a delegate call to _logic. This will typically be an encoded function call, and allows initializing the storage of the proxy like a Solidity constructor.

Source: https://docs.openzeppelin.com/contracts/4.x/api/proxy#ERC1967Proxy-constructor-address-bytes-

The proxy can initialize itself with `increment`(even if the impl. contract is initialized as a result of its constructor)  because the proxy itself is in an uninitialized state (i.e. `_initialized` is `false`).